crates/sparrow-merge/src/merge_pipeline.rs

```diff
@@ -0,0 +1,340 @@
+use arrow_schema::DataType;
+use error_stack::{IntoReport, ResultExt};
+use parking_lot::Mutex;
+use sparrow_batch::Batch;
+use sparrow_physical::StepId;
+use sparrow_scheduler::{
+    InputHandles, Partition, Partitioned, Pipeline, PipelineError, Scheduler, TaskRef,
+};
+use std::sync::atomic::{AtomicBool, Ordering};
+use tokio::sync::mpsc::error::TryRecvError;
+
+use crate::merge::HeterogeneousMerge;
+
+/// Runs a merge operation.
+pub struct MergePipeline {
+    /// The state for each partition.
+    partitions: Partitioned<MergePartition>,
+    /// Input handles for the consuming (receiving) computations.
+    consumers: InputHandles,
+    /// The left input.
+    left: Input,
```

Comment:
Should this be `Vec<Input>` to handle the multiple input case? Or is that in a follow-up?

Reply:
Follow up, added task in backlog